src/grafana_server.py

```diff
@@ -54,8 +53,8 @@ def password_has_been_changed(self, username: str, passwd: str) -> bool:
         headers = urllib3.make_headers(basic_auth="{}:{}".format(username, passwd))
 
         try:
-            self.http.request("GET", url, headers=headers)
-            return False
+            res = self.http.request("GET", url, headers=headers)
+            return True if "invalid username" in res.data.decode("utf8") else False
```

Comment:
What does it return in case the password is valid?
I.e. wouldn't it be less fragile (more future proof) if instead
```suggestion
            return False if res.data.decode("utf8") == "something very predictable" else True
```

Reply:
"Invalid username" is very specific.

If the password is valid, it returns a list of orgs, but this is not reliable as a condition depending on how grafana is provisioned -- we cannot look for any specific structure, only a success.